dfsclient/src/main/java/org/hpccsystems/dfs/client/BinaryRecordReader.java

```diff
@@ -676,6 +685,11 @@ private long getInt(int len, boolean little_endian) throws IOException
             }
         }
 
+        if (isIndex)
+        {
+            v += (0x80L << ((len-1)*8));
```

Comment:
could this be v += negMask ?
also what's the net effect of this operation?

Reply:
I could assign that value to a temporary, but it isn't a static value. In the index the essentially bias ints so they can be compared using memcpy this code returns the int back to the standard range.